RecoLocalTracker/SiPixelRecHits/src/PixelCPEFast.cc

```diff
@@ -0,0 +1,585 @@
+#include <iostream>
+
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+#include "CondFormats/SiPixelTransient/interface/SiPixelTemplate.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+#include "Geometry/TrackerGeometryBuilder/interface/RectangularPixelTopology.h"
+#include "Geometry/TrackerGeometryBuilder/interface/phase1PixelTopology.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h"
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "RecoLocalTracker/SiPixelRecHits/interface/PixelCPEFast.h"
+
+// Services
+// this is needed to get errors from templates
+
+namespace {
+  constexpr float micronsToCm = 1.0e-4;
+}
+
+//-----------------------------------------------------------------------------
+//!  The constructor.
+//-----------------------------------------------------------------------------
+PixelCPEFast::PixelCPEFast(edm::ParameterSet const& conf,
+                           const MagneticField* mag,
+                           const TrackerGeometry& geom,
+                           const TrackerTopology& ttopo,
+                           const SiPixelLorentzAngle* lorentzAngle,
+                           const SiPixelGenErrorDBObject* genErrorDBObject,
+                           const SiPixelLorentzAngle* lorentzAngleWidth)
+    : PixelCPEBase(conf, mag, geom, ttopo, lorentzAngle, genErrorDBObject, nullptr, lorentzAngleWidth, 0) {
+  EdgeClusterErrorX_ = conf.getParameter<double>("EdgeClusterErrorX");
+  EdgeClusterErrorY_ = conf.getParameter<double>("EdgeClusterErrorY");
+
+  UseErrorsFromTemplates_ = conf.getParameter<bool>("UseErrorsFromTemplates");
+  TruncatePixelCharge_ = conf.getParameter<bool>("TruncatePixelCharge");
+
+  // Use errors from templates or from GenError
+  if (UseErrorsFromTemplates_) {
+    if (!SiPixelGenError::pushfile(*genErrorDBObject_, thePixelGenError_))
+      throw cms::Exception("InvalidCalibrationLoaded")
+          << "ERROR: GenErrors not filled correctly. Check the sqlite file. Using SiPixelTemplateDBObject version "
+          << (*genErrorDBObject_).version();
+  }
+
+  // Rechit errors in case other, more correct, errors are not vailable
+  // This are constants. Maybe there is a more efficienct way to store them.
+  xerr_barrel_l1_ = {0.00115, 0.00120, 0.00088};
```

Comment:
from where does this come from!?

Reply:
answering to myself, from here: https://github.com/cms-sw/cmssw/blob/master/RecoLocalTracker/SiPixelRecHits/src/PixelCPEGeneric.cc#L101. 
Not sure this still applies anyway in both places.